src/mutation.cpp

```diff
@@ -501,6 +504,26 @@ void player::activate_mutation( const trait_id &mut )
         print_health();
         tdata.powered = false;
         return;
+    } else if( mut == trait_ROOTS2 ) {
+        double shoe_factor = footwear_factor();
+        if( g->m.has_flag( "PLOWABLE", pos() ) && shoe_factor != 1.0 && !has_effect( effect_rooted ) ) {
+            assign_activity( player_activity( activity_id( "ACT_ROOT" ), MINUTES( 5 ), -1, 0, "Rooting" ),
+                             false );
+        } else if( shoe_factor == 1.0 && !has_effect( effect_rooted ) ) {
```

Comment:
This check won't succeed the way it's currently written if the player is wearing only one shoe.

Reply:
Currently it isn't possible to wear only one shoe. Shoes are always worn in pairs. The only things that could be worn on one leg are ankle sheath, ankle ammo pouch and ankle holster, and all of them are worn on ankles (which should cover leg rather than foot, but that's IMO). Thus these ankle clothing shouldn't prevent characters from rooting.

Reply:
Looks like you're right, though it would still be good future proofing if, at some point in the future, it's possible to only wear one shoe.

The definition of `footwear_factor()` mislead me
```cpp
double player::footwear_factor() const
{
    double ret = 0;
    if( wearing_something_on( bp_foot_l ) ) {
        ret += .5;
    }
    if( wearing_something_on( bp_foot_r ) ) {
        ret += .5;
    }
    return ret;
}
```